src/Extensions/Logging/Internal/EventSource/TagsExtensions.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Globalization;
+using System.Text;
+
+namespace Microsoft.Omex.Extensions.Logging
+{
+	internal static class TagsExtensions
+	{
+		/// <summary>
+		/// Get the Tag id as a string, please avoid using this method if posible, it's added to support old tag scenario and made internal for unit tests
+		/// </summary>
+		/// <param name="tagId">tag id</param>
+		/// <returns>the tag as string</returns>
+		/// <remarks>
+		/// In terms of the conversion from integer tag value to equivalent string reprsentation, the following scheme is used:
+		/// 1. If the integer tag &lt;= 0x0000FFFF, treat the tag as special tag called numeric only tag.
+		/// Hence the string representation is direct conversion i.e. tag id 6700 == 6700
+		/// 2. Else, if it's an alphanumeric tag, there are 2 different schemes to pack those. viz. 4 letter and 5 letter representations.
+		/// 2.1 four letter tags are converted by transforming each byte into it's equivalent ASCII. e.g. 0x61626364 => abcd
+		/// 2.2 five letter tags are converted by transforming lower 30 bits of the integer value into the symbol space a-z,0-9.
+		/// The conversion is done by treating each group of 6 bits as an index into the symbol space a,b,c,d, ... z, 0, 1, 2, ....9
+		/// eg. 0x000101D0 = 00 000000 000000 010000 000111 010000 2 = aaqhq
+		/// </remarks>
+		internal static string TagIdAsString(int tagId)
```

Comment:
Can this method be marked as Obsolete?

Reply:
It suppose so, marked as obsolete, thanks